src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -0,0 +1,77 @@
+"""
+File       : MSUnmerged.py
+
+Description:
+
+This MicroService is meant to remove all the files from the unmerged area of the.
+CMS LFN Namespace. The cleanup process is supposed to happen on a per site basis.
+"""
+
+# futures
+from __future__ import division, print_function
+
+# system modules
+from retry import retry
+
+# WMCore modules
+from WMCore.MicroService.DataStructs.DefaultStructs import UNMERGED_REPORT
+from WMCore.MicroService.Unified.MSCore import MSCore
+
+# TODO:
+#       Change with the proper line for new alert mechanism.
+# from Utils.EmailAlert import EmailAlert
+
+
+class MSUnmerged(MSCore):
+    """
+    MSUnmerged.py class provides the logic for cleaning the unmerged area of
+    the CMS LFN Namespace.
+    """
+
+    def __init__(self, msConfig, logger=None):
+        """
+        Runs the basic setup and initialization for the MSUnmerged module
+        :param msConfig: micro service configuration
+        """
+        super(MSUnmerged, self).__init__(msConfig, logger=logger)
+
+        self.msConfig.setdefault("verbose", True)
+        self.msConfig.setdefault("interval", 60)
+        self.msConfig.setdefault("services", ['unmerged'])
+        self.msConfig.setdefault('limitSitesPerCycle', 100)
+        self.msConfig.setdefault('limitSitesTier', ['T1', 'T2', 'T3'])
+        self.msConfig.setdefault("rucioAccount", "FIXME_RUCIO_ACCT")
+        # self.emailAlert = EmailAlert(self.msConfig)
+
+    def execute(self):
+        """
+        Executes the whole MSUnmerged logic
+        :return: summary
+        """
+        # start threads in MSManager which should call this method
+        summary = dict(UNMERGED_REPORT)
+
+        try:
+            siteList = self.getSiteList()
+            self.updateReportDict(summary, "total_num_sites", len(requestRecords))
```

Comment:
You probably meant `len(siteList)`? BTW, I would suggest to not use the word `site` anywhere in this microservice. We are not dealing with sites, only with storage (RSEs).

Reply:
Good catch again! fixing that one too ! Thanks!

Reply:
Agreed and done